### ISSUE TITLE
Allow trigger function name customization

### ISSUE THREAD

**Issue by SeanFeldman** [Particular/NServiceBus.AzureFunctions.InProcess.ServiceBus#225]
Currently, `NServiceBusEndpointNameAttribute` allows passing in a logical endpoint name that sets the input queue name but doesn't allow customization of the trigger function. In some scenarios, the trigger function name has to follow a company deployment convention and the autogenerated `NServiceBusFunctionEndpointTrigger-{endpointName}` name doesn't work. For example:

```c#
internal const string FunctionName = "bizlogicalertmanager";
internal const string EndpointName = "bizlogic.alertmanager";
```

The desired outcome would be an ability to specify trigger function name as an optional argument to the `NServiceBusEndpointNameAttribute`. E.g.:

```c#
[assembly: NServiceBusEndpointName(EndpointName, FunctionName)]
```

**Comment by SeanFeldman** [comment]
cc @adamjones-lhpt

**Comment by adamjones-lhpt** [comment]
This is exactly our issue. We originally hosted our endpoints on cloud services, so our naming was restricted only by our transport (Azure Service Bus). We chose the endpoint name 'bizlogic.alertmanager' and our input queue was named the same. When we converted to Azure Function we met the restriction that a function name may not have a '.' character. We wanted continuity of the input queue, so we kept the endpoint / input queue name 'bizlogic.alertmanager' but changed the function name to 'bizlogicalertmanager'. With the code generator found in the 1.1.0 release of the azure function host, we are not able to specify this distinction.
